UnitsNet/CustomCode/Quantities/AmountOfSubstance.extra.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) 2013 Andreas Gullberg Larsen (andreas.larsen84@gmail.com).
+// https://github.com/angularsen/UnitsNet
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// ReSharper disable once CheckNamespace
+
+using UnitsNet.Units;
+
+namespace UnitsNet
+{
+    public partial struct AmountOfSubstance
+    {
+        public static double AvogadroConstant { get; } = 6.02214076e23;
```

Comment:
I'm adding xmldoc for this constant.

Reply:
Dude, why am I finding so many different constants on this?

- 6.02214085774 by wikipedia
- 6.022140857 by https://www.britannica.com/science/Avogadros-number
- 6.02214179 by [chem.libretexts.org](https://chem.libretexts.org/Bookshelves/Physical_and_Theoretical_Chemistry_Textbook_Maps/Supplemental_Modules_(Physical_and_Theoretical_Chemistry)/Atomic_Theory/The_Mole_and_Avogadro's_Constant)
- 6.0221415 by https://science.howstuffworks.com/avogadros-number.htm 
- 6.02214154 by https://www.scientificamerican.com/article/how-was-avogadros-number/

Reply:
I picked the one from [Wikipedia ](https://en.wikipedia.org/wiki/Avogadro_constant)due to:

> Pending revisions in the base set of SI units necessitated redefinitions of the concepts of chemical quantity. The Avogadro number, and its definition, was deprecated in favor of the Avogadro constant and its definition. Based on measurements made through the middle of 2017 which calculated a value for the Avogadro constant of NA = 6.022140758(62)×1023 mol−1, the redefinition of SI units is planned to take effect on 20 May 2019. The value of the constant will be fixed to exactly 6.02214076×1023 mol−1

[Here's ](https://www.bipm.org/utils/common/pdf/CGPM-2018/26th-CGPM-Resolutions.pdf)the reference document.

Reply:
Excellent, thanks for the reference. It's definitely worth mentioning in the xmldoc, can you add a comment about it?

Reply:
Done!